pkg/controller/planexecution/planexecution_controller.go

```diff
@@ -499,7 +499,32 @@ func (r *ReconcilePlanExecution) Reconcile(request reconcile.Request) (reconcile
 						log.Printf("Error getting patch between truth and obj: %v\n", err)
 					} else {
 						err = r.Client.Patch(context.TODO(), truth, client.ConstantPatch(types.StrategicMergePatchType, rawObj))
-						log.Printf("PlanExecutionController: CreateOrUpdate Patch: %v", err)
+						if err != nil {
+							// Right now applying a Strategic Merge Patch to custom resources does not work. There is
+							// certain metadata needed, which when missing, leads to an invalid Content-Type Header and
+							// causes the request to fail.
+							// ( see https://github.com/kubernetes-sigs/kustomize/issues/742#issuecomment-458650435 )
+							//
+							// We temporarily solve this by checking for the specific error when a SMP is applied to
+							// custom resources and handle it by defaulting to a Merge Patch.
+							//
+							// The error message for which we check is:
+							// 		the body of the request was in an unknown format - accepted media types include:
+							//			application/json-patch+json, application/merge-patch+json
+							//
+							// 		Reason: "UnsupportedMediaType" Code: 415
+							switch e := err.(type) {
+							case *errors.StatusError:
```

Comment:
Looks good to me. If you want, Kubernetes has some helper functions you can use for this: https://godoc.org/k8s.io/apimachinery/pkg/api/errors#IsUnsupportedMediaType

`if errors.IsUnsuportedMediaType(err) {}`

Reply:
> Looks good to me. If you want, Kubernetes has some helper functions you can use for this: https://godoc.org/k8s.io/apimachinery/pkg/api/errors#IsUnsupportedMediaType
> 
> `if errors.IsUnsuportedMediaType(err) {}`

Ah nice! Didn't know about this helper function!